MediaAdmin/FileUtils/Image/ImagickImageManager.php

```diff
@@ -57,10 +57,9 @@ protected function resizeImage(array $format, Imagick $image)
             $image->setimagebackgroundcolor('#000000');
             $refRatio = $maxWidth / $maxHeight;
             $imageRatio = $image->getImageWidth() / $image->getImageHeight();
-
-            if ($refRatio > $imageRatio || $maxWidth == -1) {
+            if ($imageRatio >= 1 / $maxHeight && ($refRatio > $imageRatio || $maxWidth == -1)) {
                 $image = $this->resizeOnHeight($image, $maxHeight);
-            } else {
+            } else if ($imageRatio >= 1 / $maxWidth) {
```

Comment:
@itkg-ngilain I've going on the principle that if the image do not have prerequisites to be resized, so I don't resizes the image, but I going on your idea, there will be no loss at least storage space